rclc_parameter/src/rclc_parameter/parameter_server.c

```diff
@@ -0,0 +1,618 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if __cplusplus
+extern "C"
+{
+#endif /* if __cplusplus */
+
+#include <time.h>
+
+#include <rclc_parameter/rclc_parameter.h>
+
+#include "./parameter_utils.h"
+
+#define RCLC_PARAMETER_SERVICE_MAX_LENGHT 50
+
+rcl_ret_t rclc_parameter_server_init_service(
+  rcl_service_t * service,
+  rcl_node_t * node,
+  char * service_name,
+  const rosidl_service_type_support_t * srv_type);
+
+rcl_ret_t rclc_parameter_service_publish_event(
+  rclc_parameter_server_t * parameter_server);
+
+void rclc_parameter_server_describe_service_callback(
+  const void * req,
+  void * res,
+  void * parameter_server)
+{
+  rclc_parameter_server_t * param_server = (rclc_parameter_server_t *) parameter_server;
+  DescribeParameters_Request * request = (DescribeParameters_Request *) req;
+  DescribeParameters_Response * response = (DescribeParameters_Response *) res;
+
```

Comment:
check all pointer arguments in all functions for not `NULL` pointer, like here: https://github.com/ros2/rclc/blob/64ee9e1ad02d5b3d87310cc143b69ce8440859b7/rclc/src/rclc/node.c#L29

Reply:
Addressed f111684c6414527f89c34358cdc0fe0460fc61e1